monai/utils/type_conversion.py

```diff
@@ -172,7 +172,7 @@ def convert_to_numpy(data, dtype: DtypeLike = None, wrap_sequence: bool = False)
             E.g., `[1, 2]` -> `[array(1), array(2)]`. If `True`, then `[1, 2]` -> `array([1, 2])`.
     """
     if isinstance(data, torch.Tensor):
-        data = data.detach().to(dtype=get_equivalent_dtype(dtype, torch.Tensor), device="cpu").numpy()
+        data = np.asarray(data.detach().to(device="cpu").numpy(), dtype=get_equivalent_dtype(dtype, np.ndarray))
```

Comment:
I think maybe the previous code has a better performance?
Especially when converting a GPU tensor to numpy array, it converted to the expected `dtype` on CPU directly.
But this new code makes it into 2 stages.
What do you think?

Thanks.

Reply:
this is to fix a bug https://github.com/Project-MONAI/MONAI/issues/4639